wgengine/magicsock/magicsock.go

```diff
@@ -943,6 +963,10 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 	// re-run.
 	eps = c.endpointTracker.update(time.Now(), eps)
 
+	for _, ap := range c.staticEndpoints.AsSlice() {
```

Comment:
rather than allocate here, you can:

     for i := range c.staticEndpoints.Len() {
            addAddr(c.staticEndpoints.At(i), ....)

This'll get prettier in August when Go 1.23 is out and we can use https://go.dev/doc/go1.23#language

Reply:
Nice